hasura/migrations/1626447130608_alter_table_public_player_add_column_season_xp/up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE "public"."player" ADD COLUMN "season_xp" numeric NULL DEFAULT 0;
```

Comment:
@dysbulic the total_xp table also defaults to 0 so I did it exactly like that. 
We are looking through every player in sourcecred so their xp starts off zero? 
Could you tell me a possible reason someone's xp calculation can fail?

Reply:
It would be primarily users that aren't present in Sourcecred that would have a `NULL` XP if you defaulted to `NULL`. So, it would let us know which users haven't signed up yet.

I don't know that we need that information, my point was primarily information theoretic that you're joining two sets by defaulting to `0` when keeping them separate doesn't require more work.

Hasura would return the value as `null`, and React would render it as a blank.

If you're doing calculations using the value somewhere it would add a requirement for a guard check, but all you're doing is displaying it, right?

Really though, it would make more sense to have `totalXP` be `null` to represent someone doesn't have a Sourcecred account. Since that isn't the case, I don't guess it really matters what you do for the seasonal XP.

You could make the column `NOT NULL` since there's no way it can get set to `NULL`.

Reply:
> If you're doing calculations using the value somewhere it would add a requirement for a guard check, but all you're doing is displaying it, right?

I'll be using this for sorting players on the UI and I'm a little unsure how null values will behave there @dysbulic

Reply:
Yeah I just checked and if the season_xp is null then it shows up as "highest" xp when sorting lol.